### PR TITLE
fix: safer style construction

### PR DESCRIPTION
**Pull request by pauldambra**
fixes two issues spotted in mobile replay

1) a style string was constructed incorrectly, makes the style construction safer
2) the keyboard was appearing on top of the nav bar, set explicit z-index for nav, status, and keyboard

### CODE REVIEW

File: ee/frontend/mobile-replay/transformer/screen-chrome.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+export function makeNavigationBar(
+    wireframe: wireframeDiv,
+    _children: serializedNodeWithId[],
+    context: ConversionContext
+): ConversionResult<serializedNodeWithId> | null {
+    return makeDivElement(wireframe, _children, {
+        ...context,
+        styleOverride: { ...context.styleOverride, 'z-index': TOP_OF_STACK },
+    })
+}
```
Comment (pauldambra) [inline ee/frontend/mobile-replay/transformer/screen-chrome.ts:29]: file was renamed from status-bar

adds this method for navigation bar to set an explicit z-index

File: ee/frontend/mobile-replay/transformer/screen-chrome.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        ],
+    }
+
+    return {
+        result: {
+            type: NodeType.Element,
+            tagName: 'div',
+            attributes: {
+                style: asStyleString([
+                    makeStylesString(wireframe, { 'z-index': TOP_OF_STACK }),
```
Comment (pauldambra) [inline ee/frontend/mobile-replay/transformer/screen-chrome.ts:66]: no change in this method except here

* use new safer style construction
* add an explicit z-index

File: ee/frontend/mobile-replay/transformer/transformers.ts [hunk 3]
```diff
@@ -130,7 +131,10 @@ export const makeCustomEvent = (
                     timestamp: mobileCustomEvent.timestamp,
                     idSequence: globalIdSequence,
                     skippableNodes: new Set(),
-                    styleOverride,
+                    styleOverride: {
+                        'z-index': KEYBOARD_Z_INDEX,
+                        ...(shouldAbsolutelyPosition ? {} : { bottom: true }),
+                    },
```
Comment (pauldambra) [inline ee/frontend/mobile-replay/transformer/transformers.ts:137]: place keyboard just below the nav bar but on top of everything else

File: ee/frontend/mobile-replay/__snapshots__/transform.test.ts.snap [hunk 4]
```diff
@@ -50,7 +50,7 @@ exports[`replay/transform transform can convert images 1`] = `
                   {
                     "attributes": {
                       "data-rrweb-id": 12345,
-                      "style": "color: #ffffff;width: 100px;height: 30px;position: fixed;left: 11px;top: 12px;overflow:hidden;white-space:nowrap;",
+                      "style": ";;color: #ffffff;width: 100px;height: 30px;position: fixed;left: 11px;top: 12px;overflow:hidden;white-space:nowrap;",
```
Comment (pauldambra) [inline ee/frontend/mobile-replay/__snapshots__/transform.test.ts.snap:53]: well, that's not right 🫠

File: ee/frontend/mobile-replay/__snapshots__/transform.test.ts.snap [hunk 5]
```diff
@@ -850,34 +850,7 @@ exports[`replay/transform transform incremental mutations de-duplicate the tree
...
-AIA8gQMAAADIEzgAAACAPIEDAAAAyBM4AAAAgDyBAwAAAMgTOAAAAIA8gQMAAADIEzgAAACAPIED
-AAAAyBM4AAAAgDyBAwAAAMgTOAAAAIA8gQMAAADIEzgAAACAPIEDAAAAyBM4AAAAgDyBAwAAAMgT
-OAAAAIA8gQMAAADIEzgAAACAPIEDAAAAyBM4AAAAgDyBAwAAAMgTOAAAAIA8gQMAAADIEzgAAACA
-PIEDAAAAyBM4AAAAgDyBAwAAAMgTOAAAAIA8gQMAAADIEzgAAACAPIEDAAAAyBM4AAAAgDyBAwAA
-AMgTOAAAAIA8gQMAAADIEzgAAACAPIEDAAAAyBM4AAAAgDyBAwAAAMgTOAAAAIA8gQMAAADIEzgA
-AACAPIEDAAAAyBM4AAAAgDyBAwAAAMgTOAAAAIA8gQMAAADIOwscL/6XFQAAAADHDhvFWeD47YJD
-AAAAAN7XYaNYRw/33p/PzC/jUxYAAADg9rydmS/WWk+vOjgMF2utJzPz+NKrAAAAAK7h0VHcmDl5
-g+Nve+8vZ+b7mflkZu5dYBgAAADAkT9m5vnMPFxr/XrbYwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
-AAAAAAAAAAAAAAAAAAAAAAAAgCN/AW0xMqHnNQceAAAAAElFTkSuQmCC
-);background-size: auto;background-repeat: no-repeatwidth: 411px;height: 70px;position: fixed;left: 0px;top: 536px;overflow:hidden;white-space:nowrap;",
```
Comment (pauldambra) [inline ee/frontend/mobile-replay/__snapshots__/transform.test.ts.snap:880]: `background-repeat: no-repeatwidth:`

you can see the missing semi-colon between repeat and width here

**Review by daibhin — APPROVED** [review]

**Review by marandaneto — APPROVED** [review]

**Pull request merged**